### PR TITLE
Binary Search added. Recurive implementation added. Tests added

### PR DESCRIPTION
**Pull request by soummyaah**
Please check and approve.
Refs #12 & #4 

- l1ghtsab3r Team

### CODE REVIEW

**Comment by aktech** [comment]
@soummyaah  Add the reference to these function in docs.

**Comment by aktech** [comment]
I think you can wrap both these functions in a single function as:

```
def binary_search(item_list, item, recursive=False)
```
When the keyword argument recursive is true, it will do a recursive binary search otherwise it will do a iterative binary search.

File: docs/source/searching.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.. _sorting:
+
+Searching
+=======
+
+Sorting is a process of finding an item from a list. This process can be implemented via many different algorithms. Following is the list of searching algorithms which are implemented in Pydsa.
```
Comment (malayaleecoder) [inline docs/source/searching.rst]: "Sorting" must be changed to "Searching"

Comment (soummyaah) [inline docs/source/searching.rst]: Right!
Thanks for spotting it :D

File: pydsa/binary_search.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+def binary_search(items_list, item, recursive=False, first_ptr = -1, last_ptr = -1):
```
Comment (aktech) [inline pydsa/binary_search.py]: Please follow PEP8 Standards:
pep8: E251 remove space around `=`
i.e. `first_ptr = -1` --> `first_ptr=1` & `last_ptr = -1` --> `last_ptr=-1`

Tip: there is a plugin for checking pep8 in every code editor, checkout for your code editor.

**Comment by soummyaah** [comment]
@aktech I've fixed the pep errors. Let me know what else needs to be changed to be merged.

File: docs/source/searching.rst [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.. _sorting:
+
+Searching
+=======
+
+Searching is a process of finding an item from a list. This process can be implemented via many different algorithms. Following is the list of searching algorithms which are implemented in Pydsa.
```
Comment (aktech) [inline docs/source/searching.rst]: Pydsa --> PyDSA

File: pydsa/binary_search.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    """
+    Searches the list 'items_list' for an item 'item'
+    using binary search algorithm
+
+    >>> from pydsa import binary_search
+    >>> a = [1, 2, 7, 9, 10, 33, 56, 70, 99]
+    >>> binary_search(a, 9)
+    3
+    >>> binary_search(a, 13)
+    -1
```
Comment (aktech) [inline pydsa/binary_search.py]: Add an example for recursive also.

**Comment by aktech** [comment]
> I've fixed the pep errors. 

Excellent! The PR is ready to be merged now, except for a couple of minor comments.

**Comment by soummyaah** [comment]
@aktech Done! :D

**Comment by aktech** [comment]
@soummyaah Thanks! Its in.

**Pull request merged**